internal/metrics/metrics.go

```diff
@@ -187,6 +227,172 @@ func GetScalingSignal(pdb *policyv1.PodDisruptionBudget) string {
 	return PDBBlockedSignal
 }
 
+// GetScalingOpportunityLabels gets age pattern and effectiveness labels for scaling opportunities
+func GetScalingOpportunityLabels(ctx context.Context, c client.Client, pdb *policyv1.PodDisruptionBudget) (string, string) {
```

Comment:
probabluy want a new file if not a new pacakge for AnalyzePodAgeFailurePattern and mabye GetScalingOpportunityLabels